.github/actions/docker-build-push/action.yml

```diff
@@ -0,0 +1,50 @@
+name: docker build and push
+description: builds and pushes a docker image to docker registry
+inputs:
+  DOCKER_REG_USERNAME:
+    required: true
+  DOCKER_REG_PASSWORD:
+    required: true
+  dockerfile-path:
+    required: false
+    description: uses root dir if not specified
+  image-name:
+    required: true
+  build-args:
+    required: false
+outputs:
+  imageid:
+    value: ${{ steps.build-and-push.outputs.imageid }}
+runs:
+  using: "composite"
+  steps:
+    - name: Login to DockerHub
+      uses: docker/login-action@v2
+      with:
```

Comment:
can we add a registry type here as well. In our case it is dockerhub but could be gcr or any other in certain cases.

Reply:
registry added as a input, if no input is provided then dockerhub will be considered by default